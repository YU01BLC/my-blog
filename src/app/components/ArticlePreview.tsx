import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import CardHeader from '@mui/material/CardHeader';
import CardMedia from '@mui/material/CardMedia';
import Typography from '@mui/material/Typography';
import MoreDescription from './ArticleDetail/MoreDescription';

export default function DummyArticle({ params: { id } }: { params: { id: string } }) {
  console.log(id);
  return (
    <Card data-testid='dummy-article'>
      <CardHeader title='記事タイトル(dummy)' subheader='2023/12/1' />

      <CardMedia
        component='img'
        image={'https://source.unsplash.com/collection/1346951/1000x500?sig=2'}
        alt='blog image'
        width='100%'
        height='500'
      />
      <CardContent>
        <Typography variant='body1' color='text.secondary' sx={{ paddingTop: '1em' }}>
          ダミー記事です。
          <br />
          右上の「記事を書く」ボタンから新規記事を執筆してください。
        </Typography>
        <Typography variant='body1' color='text.secondary' sx={{ paddingTop: '1em' }} data-testid='dummy-article--text'>
          ※ &thinsp; 記事投稿後ダミー記事は非表示となります。
        </Typography>
        <MoreDescription />
      </CardContent>
    </Card>
  );
}
