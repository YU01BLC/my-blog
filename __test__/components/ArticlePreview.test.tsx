import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import ArticlePreview from '../../src/app/components/ArticlePreview';

describe('DummyArticleコンポーネントSnapshot', () => {
  it('snapshot', () => {
    const testData = { id: 'testId' };
    const { asFragment } = render(<ArticlePreview params={testData} />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('DummyArticleコンポーネント表示確認', () => {
  it('DummyArticleコンポーネントが表示されていること', () => {
    const testData = { id: 'testId' };
    render(<ArticlePreview params={testData} />);
    const element = screen.getByTestId('dummy-article');
    expect(element).toBeInTheDocument();
  });
});

describe('Cardコンポーネント表示確認', () => {
  it('CardHeaderコンポーネントが表示されていること', () => {
    const testData = { id: 'testId' };
    render(<ArticlePreview params={testData} />);
    const cardElement = screen.getByText('記事タイトル(dummy)');
    expect(cardElement).toBeInTheDocument();
  });
  it('CardMediaコンポーネントが表示されていること', () => {
    const testData = { id: 'testId' };
    render(<ArticlePreview params={testData} />);
    const cardElement = screen.getByAltText(/blog image/i);
    expect(cardElement).toBeInTheDocument();
  });
  it('Typographyコンポーネントが表示されていること', () => {
    const testData = { id: 'testId' };
    render(<ArticlePreview params={testData} />);
    const cardElement = screen.getByTestId('dummy-article--text');
    expect(cardElement).toBeInTheDocument();
  });
});
