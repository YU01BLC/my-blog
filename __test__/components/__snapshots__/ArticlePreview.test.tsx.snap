// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DummyArticleコンポーネントSnapshot snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    data-testid="dummy-article"
  >
    <div
      class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
    >
      <div
        class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
        >
          記事タイトル(dummy)
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiCardHeader-subheader css-nrdprl-MuiTypography-root"
        >
          2023/12/1
        </span>
      </div>
    </div>
    <img
      alt="blog image"
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
      height="500"
      src="https://source.unsplash.com/collection/1346951/1000x500?sig=2"
      width="100%"
    />
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-136xlp0-MuiTypography-root"
      >
        ダミー記事です。
        <br />
        右上の「記事を書く」ボタンから新規記事を執筆してください。
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-136xlp0-MuiTypography-root"
        data-testid="dummy-article--text"
      >
        ※   記事投稿後ダミー記事は非表示となります。
      </p>
      <div
        class="MuiCardActions-root css-1rwjz6-MuiCardActions-root"
      >
        <div
          aria-expanded="false"
          aria-label="show more"
          class="MuiBox-root css-d3fn1e"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            続きを読む
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
